internal/statemachine/classic_states.go

```diff
@@ -658,88 +658,78 @@ func (stateMachine *StateMachine) preseedClassicImage() error {
 // populateClassicRootfsContents copies over the staged rootfs
 // to rootfs. It also changes fstab and handles the --cloud-init flag
 func (stateMachine *StateMachine) populateClassicRootfsContents() error {
-	/*	var classicStateMachine *ClassicStateMachine
-		classicStateMachine = stateMachine.parent.(*ClassicStateMachine)
-
-		var src string
-		if classicStateMachine.Opts.Filesystem != "" {
-			src = classicStateMachine.Opts.Filesystem
-		} else {
-			src = filepath.Join(classicStateMachine.tempDirs.unpack, "chroot")
-		}
+	var classicStateMachine *ClassicStateMachine
+	classicStateMachine = stateMachine.parent.(*ClassicStateMachine)
 
-		files, err := ioutilReadDir(src)
-		if err != nil {
-			return fmt.Errorf("Error reading unpack/chroot dir: %s", err.Error())
-		}
+	src := stateMachine.tempDirs.chroot
 
-		for _, srcFile := range files {
-			srcFile := filepath.Join(src, srcFile.Name())
-			if err := osutilCopySpecialFile(srcFile, classicStateMachine.tempDirs.rootfs); err != nil {
-				return fmt.Errorf("Error copying rootfs: %s", err.Error())
-			}
-		}
+	files, err := ioutilReadDir(src)
+	if err != nil {
+		return fmt.Errorf("Error reading unpack/chroot dir: %s", err.Error())
+	}
 
-		fstabPath := filepath.Join(classicStateMachine.tempDirs.rootfs, "etc", "fstab")
-		fstabBytes, err := ioutilReadFile(fstabPath)
-		if err == nil {
-			if !strings.Contains(string(fstabBytes), "LABEL=writable") {
-				re := regexp.MustCompile(`(?m:^LABEL=\S+\s+/\s+(.*)$)`)
-				newContents := re.ReplaceAll(fstabBytes, []byte("LABEL=writable\t/\t$1"))
-				if !strings.Contains(string(newContents), "LABEL=writable") {
-					newContents = []byte("LABEL=writable   /    ext4   defaults    0 0")
-				}
-				err := ioutilWriteFile(fstabPath, newContents, 0644)
-				if err != nil {
-					return fmt.Errorf("Error writing to fstab: %s", err.Error())
-				}
-			}
+	for _, srcFile := range files {
+		srcFile := filepath.Join(src, srcFile.Name())
+		if err := osutilCopySpecialFile(srcFile, classicStateMachine.tempDirs.rootfs); err != nil {
+			return fmt.Errorf("Error copying rootfs: %s", err.Error())
 		}
+	}
 
-		if classicStateMachine.commonFlags.CloudInit != "" {
-			seedDir := filepath.Join(classicStateMachine.tempDirs.rootfs, "var", "lib", "cloud", "seed")
-			cloudDir := filepath.Join(seedDir, "nocloud-net")
-			err := osMkdirAll(cloudDir, 0756)
-			if err != nil && !os.IsExist(err) {
-				return fmt.Errorf("Error creating cloud-init dir: %s", err.Error())
+	fstabPath := filepath.Join(classicStateMachine.tempDirs.rootfs, "etc", "fstab")
+	fstabBytes, err := ioutilReadFile(fstabPath)
+	if err == nil {
```

Comment:
Shouldn't we only tweak the `fstab` *if* the user did not specify a custom fstab in the image-definition? I think it makes sense that if the user did specify an explicit fstab, they know best what contents are expected there.